api/handlers.go

```diff
@@ -865,27 +941,33 @@ func (api *API) userRecoverGET(w http.ResponseWriter, req *http.Request, _ httpr
 // They need to provide a valid password-reset token.
 // The user doesn't need to be logged in.
 func (api *API) userRecoverPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	if err := req.ParseForm(); err != nil {
-		api.WriteError(w, err, http.StatusBadRequest)
+	defer api.closeBody(req)
+	// Parse the request's body.
+	var payload accountRecoveryDTO
+	b, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to parse request body"), http.StatusBadRequest)
+		return
+	}
+	err = json.Unmarshal(b, &payload)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to parse request body"), http.StatusBadRequest)
```

Comment:
update error message

Reply:
Same as above, changed the message of the preceding error.